### ISSUE TITLE
Dense vector recall-10-100 scores are weird

### ISSUE THREAD

**Issue by benwtrent** [elastic/rally-tracks#463]
After: https://github.com/elastic/rally-tracks/pull/454

It seems that `recall-10-100` & `recall-100-1000` is actually reporting the multi-segment recall numbers and not the single segment recall numbers: 

See:
https://elasticsearch-benchmarks.elastic.co/#tracks/dense_vector/nightly/default/30d

`nightly-dense_vector-avg-recall`

These additions:
```
{
      "name": "knn-recall-10-100_multiple_segments",
      "operation": "knn-recall-10-100"
    },
    {
      "name": "knn-recall-100-1000_multiple_segments",
      "operation": "knn-recall-100-1000"
    },
```
Are effecting the averages. I am wondering if we are not purging the averages or mis-naming things in the operation parameters?

**Comment by benwtrent** [comment]
fyi @jdconrad && @mayya-sharipova

**Comment by mayya-sharipova** [comment]
@benwtrent Thanks for letting us know, I will investigate

**Comment by benwtrent** [comment]
@mayya-sharipova I think its just a query visualization thing. Take a look at this:
```
"bool": {
            "must": [
              {
                "query_string": {
                  "query": "task:*recall*",
                  "analyze_wildcard": true
                }
              }
            ],
            "filter": [],
            "should": [],
            "must_not": []
          }
        }
```

**Comment by gareth-ellis** [comment]
I think its because we grouping by operation, which is the same for both tasks: 
<img width="1015" alt="image" src="https://github.com/elastic/rally-tracks/assets/14981026/782a77e9-6308-4e00-8739-68a93333acc7">

**Comment by mayya-sharipova** [comment]
@gareth-ellis Thanks for the investigation. Looks like we need to modify dashboard to group by task?

**Comment by gareth-ellis** [comment]
Yes, it seems so - i have it open so can change that now, if you would like?

**Comment by benwtrent** [comment]
Just need to not-match the `*_multiple_segments` when matching`task:*recall*` :) Thank you @gareth-ellis

**Comment by benwtrent** [comment]
@mayya-sharipova yeah, we could group_by `task` instead of by `operation`. I am fine with that.

**Comment by gareth-ellis** [comment]
Updated, take a look :)

**Comment by mayya-sharipova** [comment]
@gareth-ellis Thank you for the update. I can't still see the changes yet, when they will be publicly available?

**Comment by gareth-ellis** [comment]
> I mean changes to see recall for `multiple_segments`.

Ah, now it should be there

**Comment by benwtrent** [comment]
@gareth-ellis perfect!

<img width="689" alt="image" src="https://github.com/elastic/rally-tracks/assets/4357155/f24cd2b5-c78f-4c5e-933a-94c18a08af18">
